rust/cymbal/src/issue_resolution.rs

```diff
@@ -0,0 +1,143 @@
+        Some(f) => f,
+        None => create_issue(pool, team_id, fingerprint).await?,
+    };
+
+    Ok(issue_fingerprint.issue_id)
+}
+
+pub async fn load_issue_override<'c, E>(
+    executor: E,
+    team_id: i32,
+    fingerprint: &str,
+) -> Result<Option<IssueFingerprintOverride>, Error>
+where
+    E: sqlx::Executor<'c, Database = sqlx::Postgres>,
+{
+    Ok(sqlx::query_as!(
+        IssueFingerprintOverride,
+        r#"
+            SELECT id, team_id, fingerprint, issue_id, version
+            FROM posthog_errortrackingissuefingerprintv2
+            WHERE team_id = $1 AND fingerprint = $2
+        "#,
+        team_id,
+        fingerprint
+    )
+    .fetch_optional(executor)
+    .await?)
+}
+
+pub async fn create_issue<'c, A>(
+    connection: A,
+    team_id: i32,
+    fingerprint: &str,
+) -> Result<IssueFingerprintOverride, Error>
+where
+    A: sqlx::Acquire<'c, Database = sqlx::Postgres>,
+{
+    let issue_id = Uuid::now_v7();
+
+    let mut tx = connection.begin().await?;
+
+    sqlx::query!(
+        r#"
+            INSERT INTO posthog_errortrackingissue (id, team_id)
+            VALUES ($1, $2)
+        "#,
+        issue_id,
+        team_id
+    )
+    .execute(&mut *tx)
+    .await?;
+
+    let res = sqlx::query_as!(
+        IssueFingerprintOverride,
+        r#"
+            INSERT INTO posthog_errortrackingissuefingerprintv2 (id, team_id, fingerprint, issue_id, version)
+            VALUES ($1, $2, $3, $4, 0)
+            ON CONFLICT (team_id, fingerprint) DO UPDATE SET version = posthog_errortrackingissuefingerprintv2.version + 1
+            RETURNING id, team_id, fingerprint, issue_id, version
+        "#,
+        Uuid::now_v7(),
+        team_id,
+        fingerprint,
+        issue_id
+    )
+    .fetch_one(&mut *tx)
+    .await?;
+
+    tx.commit().await?;
+
+    // TODO: write to Kafka
+
+    Ok(res)
+}
+
+#[cfg(test)]
+mod test {
+    use sqlx::PgPool;
+
+    #[sqlx::test(migrations = "./tests/test_migrations")]
+    async fn test_issue_creation(db: PgPool) {
+        let team_id: i32 = 1;
+        let fingerprint = "this_is_a_fingerprint".to_string();
+
+        super::resolve_issue_id(&db, team_id, &fingerprint)
+            .await
+            .unwrap();
+
+        // Verify both records are created in Postgres
+        let record = super::load_issue_override(&db, team_id, &fingerprint)
+            .await
+            .unwrap()
+            .unwrap();
+
+        assert_eq!(record.fingerprint, fingerprint);
+        assert_eq!(record.version, 0);
+
+        super::resolve_issue_id(&db, team_id, &fingerprint)
+            .await
+            .unwrap();
+
+        let result = sqlx::query!(
+            r#"
+                SELECT COUNT(*)
+                FROM posthog_errortrackingissue
+            "#,
+        )
+        .fetch_one(&db)
+        .await
+        .unwrap();
+
+        // Only a single issue is created
+        assert_eq!(result.count, Some(1))
```

Comment:
@oliverb123 any idea why this needs to be wrapped in `Some`?

Reply:
😕  feels related to https://github.com/launchbadge/sqlx/issues/3286